src/vs/base/test/common/map.test.ts

```diff
@@ -588,7 +588,7 @@ suite('Map', () => {
 	test('mapToSerializable / serializableToMap', function () {
 		const map = new Map<string, string>();
 		map.set('1', 'foo');
-		map.set('2', null);
+		map.set('2', null!);
```

Comment:
what does `null!` do? can't seem to wrap my head around it

Reply:
Because of `strict-null-check` mode, the `null` and `string` are not in the domain of the same type. It will throw error when typescript compile with `--strictNullChecks`. You can know `strict-null-check` in the [Typescript Doc](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html).

Reply:
I understand the non null assertion operator generally, for example when used with objects, it indicates to the compiler that we are sure the object is not going to be null even if the compiler can't infer it.
 i just don't understand why is it used with null here and what does it mean to the compiler in this case.

We are passing null to something which is not declared as nullable, and cheating the compiler by adding a ! ?